tests/boardfarm_plugins/boardfarm_prplmesh/tests/client_association_link_metrics.py

```diff
@@ -19,16 +20,20 @@ class ClientAssociationLinkMetrics(PrplMeshBaseTest):
     Response message containing an Associated STA Link Metrics TLV for the associated STA.'''
 
     def runTest(self):
-        # Locate test participants in array
-        for dev in self.dev:
-            if dev.controller_entity:
-                controller = dev.controller_entity
-            if dev.agent_entity:
-                agent = dev.agent_entity
+        # Locate test participants
+        controller = self.dev.DUT.controller_entity
+        agent = self.dev.lan.agent_entity
+        sta = self.dev.wifi
+
+        # This test doesn't work for real HW.
+        # Skip it for prplWRT device.
+        # TODO: Remove as soon, as test works for prplWRT device
+        if self.dev.DUT is PrplMeshPrplWRT:
+            self.skipTest("This test isn't ready for prplWRT devices")
```

Comment:
Could we instead do something like
```
if not self.dev.has_wifi:
    self.skipTest(...)
```

Or maybe:

```
try:
    sta = self.dev.wifi
except AttributeError:
    self.skipTest(...)
```

Note that the latter could be factored in a setup function. Let's say that PrplMeshBaseTest has a setup method that looks a bit like this:
```
def setUp(self):
    try:
        self.configure()
    except AttributeError, e:
        self.skipTest(... use e to find which device was missing ...)
```

Alternatively, we could add a get_device function that raises the skipTest if needed.